xds/src/main/java/io/grpc/xds/sds/FileBasedPluginCredential.java

```diff
@@ -90,6 +91,12 @@ private static DataSource buildDataSourceFromConfigStruct(Struct secretValueStru
   @Override
   public void applyRequestMetadata(
       @Nullable RequestInfo requestInfo, Executor appExecutor, final MetadataApplier applier) {
+    if (!allowedSecurityLevel(requestInfo, SecurityLevel.PRIVACY_AND_INTEGRITY)) {
```

Comment:
Actually, this may not be correct. @sanjaypujare, can you comment whether we may be required to use bearer tokens on an unauthenticated connection?

Reply:
If you are asking about "unauthenticated", yes - with at least 2 examples :

- in case of SDS over UDS the connection does not authenticate the client but we do send bearer tokens (as in this code) over that channel

- in TLS  (non-mTLS) the client is not authenticated to the server but the client can send bearer tokens.

IIRC both UDS and TLS are considered at `SecurityLevel.PRIVACY_AND_INTEGRITY` in this PR to support the above cases. We probably should have more levels to accurately reflect some of the characteristics.

Reply:
For plain TLS it is fine that the client is unauthenticated; the client doesn't need to authenticate itself, just the server. If FileBasedPluginCredential is used for SDS, then this would break it currently. Right now UDS is not considered PRIVACY_AND_INTEGRITY; there's no special code for it so it is the same as plain-text.

Reply:
Above in https://github.com/grpc/grpc-java/pull/6616#issuecomment-575827581 @yihuazhang did say this "...I have a following plan to associate the security level PRIVACY_AND_INTEGRITY to a UDS channel...". Haven't heard anything since then... @yihuazhang can you throw some light on this?

Reply:
But that isn't implemented. It seems we should not add the check here, to let this PR be merged.

Reply:
Yes, I agree. Unless he decides to implement that other thing in this PR

Reply:
we agreed in go/grpc-security-level-negotiation that UDS is treated as privacy-and-integrity.

Reply:
So that should be implemented here, right?

Reply:
@jiangtaoli2016, I'm not arguing that UDS _shouldn't_ be privacy-and-integrity. I'm arguing that it _isn't_ today. So adding this code would completely break the SDS implementation. Adding this code is also unrelated to the goal of this PR: to add a utility function for credential implementations. If that is the goal, then we should drop the change in this file and merge the utility.

Reply:
Thanks all for the clarification. I will drop the changes in this function and will re-visit it in a separate PR (maybe after UDS is supported and a correct security level is set for it).

Reply:
The checks are already in in some places like https://github.com/grpc/grpc-java/blob/master/auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java#L93 . Need to make sure those won't break anything.

Reply:
If the check is already there, then using the utility is safe. They produce the same results.